### PR TITLE
VAULT-32410: upgrade go lang crypto and other go lang libraries

### PR DESCRIPTION
**Pull request by aslamovamir**
# Overview
This PR upgrades the golang.org/x/crypto library as well as some other go lang libraries. This is needed to address a security concern from https://hashicorp.atlassian.net/browse/SECVULN-14980

# Related Issues/Pull Requests
- [VAULT-32410](https://hashicorp.atlassian.net/browse/VAULT-32410)

# Contributor Checklist
- This change should not affect user behavior.

[VAULT-32410]: https://hashicorp.atlassian.net/browse/VAULT-32410?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -58,9 +58,9 @@ require (
 	go.mongodb.org/mongo-driver v1.12.0 // indirect
 	go.opentelemetry.io/otel v1.14.0 // indirect
 	go.opentelemetry.io/otel/trace v1.14.0 // indirect
-	golang.org/x/crypto v0.15.0 // indirect
-	golang.org/x/net v0.18.0 // indirect
-	golang.org/x/sys v0.14.0 // indirect
+	golang.org/x/crypto v0.29.0 // indirect
+	golang.org/x/net v0.21.0 // indirect
```
Comment (VioletHynes) [inline go.mod:62]: I'd recommend we upgrade this to the latest version (0.31.0)

**Review by VioletHynes — APPROVED** [review]
Looks good, with one minor comment :)

File: go.mod (hunk 1 above)
Comment (aslamovamir) [reply go.mod]: Upgraded that as well, thanks!

**Comment by aslamovamir** [comment]
Need one more review from vault-core-hybrid team members, @divyaac or @biazmoreira can you please take a look when you have time? :) Thank you!

**Review by divyaac — APPROVED** [review]
Looks great! thanks Amir :)

**Pull request merged**